spec/user_interface_spec.rb

```diff
@@ -22,6 +23,34 @@
     end
   end
 
+  describe 'who_goes_first?' do
+    it 'asks player who should go first' do
+      allow(user_interface).to receive(:gets).and_return('1')
+      expect(STDOUT).to receive(:puts).with(first_player)
+      user_interface.who_goes_first?
+    end
+
+    it 'returns the user selection' do
+      allow(STDOUT).to receive(:puts)
+      allow(user_interface).to receive(:gets).and_return('1')
+      allow(STDOUT).to receive(:puts)
+      expect(user_interface.who_goes_first?).to eql('1')
+      allow(STDOUT).to receive(:puts).twice
+      allow(user_interface).to receive(:gets).and_return('2')
+      expect(user_interface.who_goes_first?).to eql('2')
```

Comment:
OK, changed who_goes_first and what_mode to call process_input which takes the callback get_input.  I did it in the other branch since i made a lot of changes in the ui there.

[Updated UI](https://github.com/saralein/tic-tac-toe_ruby/pull/7/files#diff-2bad3e77eef91be4afbbff5bceebfe6dR13)
[Updated Test](https://github.com/saralein/tic-tac-toe_ruby/pull/7/files#diff-7aa5a624018bd299e043383e4af27151R35)